### DISCUSSION TITLE
Possibility to extract and match on file meta-data

### DISCUSSION THREAD

**Discussion by forgedhallpass** [projectdiscovery/nuclei#1504] [Ideas]
The file protocol could be extended to retrieve additional meta-data of the target files. The data could be made available on the template context, so that matchers and extractors could be applied over them.

Such metadata could be:
- EXIF data of images (could be used for OSINT)
- password protection, certificates, general description containing (sub)domains, e-mail addresses of PDF and other binary document files
- calculated hash of the file. It could be used to identify file dependencies (e.g. JARs) that contain known vulnerabilities
- etc.

In order to decide on implementing such functionalities, we need describe potential use-cases.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
